mjordan/islandora_workbench
# Add term support for entity_reference_view_endpoints

## Link to Github issue or other discussion

#642

## What does this PR do?

Uses the configured endpoints for term lookups.

## What changes were made?

* Adds a check if the entity_reference_view_endpoints config is set up
* Uses the configured endpoint for a given field to perform the lookup
* If the term is not found and a vocabulary is provided, create the term
* Additional check provides a warning and omits the field value if nothing is found or created

## How to test / verify this PR?

Set up entity_reference_view_endpoints in your config file, e.g.
```yml
entity_reference_view_endpoints:
  field_linked_agent: /taxonomy/linked_agents
  field_language: /language/lookup
  field_subject: /taxonomy/subjects
```
See https://github.com/mjordan/islandora_workbench/issues/452#issuecomment-1577523240

Perform a create or update action with fields with and without the term vocabulary that do and don't exist and see the appropriate log messages appear.

E.g.
```csv
node_id,field_linked_agent
118116,relators:pbl:Arizona State University
118117,relators:pbl:corporate_body:Arizona State University
118118,relators:pbl:DOES NOT EXIST
118119,relators:pbl:corporate_body:DOES NOT EXIST
```
## Interested Parties

@mjordan

---

## Checklist

* [:heavy_check_mark:] Before opening this PR, have you opened an issue explaining what you want to to do?
* [:heavy_check_mark:] Have you run `pycodestyle --show-source --show-pep8 --ignore=E402,W504 --max-line-length=200 yourfile.py`? 
* [:heavy_check_mark:] Have you included some configuration and/or CSV files useful for testing this PR?
* [ No ] Have you written unit or integration tests if applicable?
* [ No ] Does the code added in this PR require a version of Python that is higher than the current minimum version?
* [ N/A ] If the changes in this PR require an additional Python library, have you included it in `setup.py`?
* [ N/A ] If the changes in this PR add a new configuration option, have you provided a default for when the option is not present in the .yml file?
